src/dropdown/dropdown.js

```diff
@@ -170,9 +162,50 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
         return;
       }
       dropdownCtrl.dropdownMenu = element;
+
+      var dropdownMenu = element,
+          dropdown = element.parent(),
```

Comment:
@wesleycho Makes sense. I've just pushed changes to accommodate these user cases (test included).